### PR TITLE
[JBPM-9426] No tasks returned when userId == null, exlOwner !=null and groups !=null

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9426

### CODE REVIEW

**Comment by gmunozfe** [comment]
jenkins retest this

**Comment by elguardian** [comment]
Jenkins retest this please.

**Review by gmunozfe — CHANGES_REQUESTED** [review]
I think same change is required for other queries:
- TasksAssignedAsPotentialOwnerWithGroups
- TasksAssignedAsPotentialOwnerByStatusWithGroups
- TasksAssignedAsPotentialOwnerStatusByExpirationDate
- TasksAssignedAsPotentialOwnerStatusByExpirationDateOptional
- SubTasksAssignedAsPotentialOwner
- QuickTasksAssignedAsPotentialOwnerWithGroupsByStatus
- QuickTasksAssignedAsPotentialOwnerStatusByExpirationDateOptional
- QuickTasksAssignedAsPotentialOwnerStatusByExpirationDate
- NewTasksAssignedAsPotentialOwner
- NewTasksAssignedAsPotentialOwnerByExpirationDateOptional
- NewTasksAssignedAsPotentialOwnerByExpirationDate

**Comment by elguardian** [comment]
good catch @gmunozfe

**Comment by elguardian** [comment]
Jenkin retest this please.

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java [hunk 1]
```diff
@@ -532,7 +768,7 @@ public void testGetTasksAssignedByGroupsWithGroupsLangNoTask() {
     
     @Test
     public void testGetTasksAssignedByGroupsWithGroupsLangOneTask() {
-        // One potential owner, should go straight to state Reserved
+        // One potential owner, should go straight to s)tate Reserved
```
Comment (gmunozfe) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:771]: typo

**Review by gmunozfe — APPROVED** [review]
Looks good to me, nailed down @elguardian !

**Comment by gmunozfe** [comment]
jenkins retest this

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java [hunk 2]
```diff
@@ -118,21 +119,20 @@ public void taskCmdUniqueRootElementTest() throws Exception {
 
     @Test
     public void taskCommandSubTypesCanBeSerialized() throws Exception {
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java:121]: No longer throwing a checked exception

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java [hunk 3]
```diff
@@ -118,21 +119,20 @@ public void taskCmdUniqueRootElementTest() throws Exception {
...
+        reflections.getSubTypesOf(TaskCommand.class).stream().filter(e -> !e.getName().contains("$")).forEach(jaxbClass ->  {
+            if (jaxbClass.equals(UserGroupCallbackTaskCommand.class) || jaxbClass.equals(GetCurrentTxTasksCommand.class)) {
+                return;
             }
-            addClassesToSerializationContext(jaxbClass);
-            Constructor<?> construct = jaxbClass.getConstructor(new Class[] {});
             try {
+                addClassesToSerializationContext(jaxbClass);
+                Constructor<?> construct = jaxbClass.getConstructor(new Class[] {});
                 Object jaxbInst = construct.newInstance(new Object[] {});
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java:129]: Not really needed
```suggestion
                Constructor<?> construct = jaxbClass.getConstructor();
                Object jaxbInst = construct.newInstance();
```

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java [hunk 4]
```diff
@@ -23,21 +23,29 @@
...
+import org.jbpm.services.task.utils.ClassUtil;
 import org.junit.Ignore;
 import org.junit.Test;
+import org.kie.api.command.ExecutableCommand;
+import org.kie.api.runtime.Context;
 import org.kie.api.task.model.I18NText;
 import org.kie.api.task.model.Status;
 import org.kie.api.task.model.Task;
 import org.kie.api.task.model.TaskSummary;
+import org.kie.internal.executor.api.Command;
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:48]: Please remove unused imports

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java [hunk 5]
```diff
@@ -270,6 +278,234 @@ public void testGetTasksAssignedAsPotentialOwnerWithUserGroupsLangStatus() {
...
+        List<TaskSummary> tasks = taskService.execute(new TaskCommand<List<TaskSummary>> () {
+
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public List<TaskSummary> execute(Context cntxt) {
+                TaskContext context = (TaskContext) cntxt;
+                
+                return (List<TaskSummary>) context.getPersistenceContext()
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:498]: Redundant cast
```suggestion
                return context.getPersistenceContext()
```

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java [hunk 6]
```diff
@@ -270,6 +278,234 @@ public void testGetTasksAssignedAsPotentialOwnerWithUserGroupsLangStatus() {
     }
     
     
+    @Test
+    public void testGetTasksAssignedAsPotentialOwnerWithNoUserGroupsLangStatus() {
+        startBaseTest();
+        final List<String> groups = new ArrayList<String>();
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:284]: Explicit type argument not really needed
```suggestion
        final List<String> groups = new ArrayList<>();
```

**Review by afalhambra — APPROVED** [review]
Very good job

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java (hunk 2 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java:121]: removed

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java (hunk 3 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/jaxb/JaxbTaskSerializationTest.java:129]: jaxbClass.newInstance

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java (hunk 4 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:48]: removed

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java (hunk 5 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:498]: removed

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java (hunk 6 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java:284]: removed

**Pull request merged**